Scan Importer/Qualys infra scan WebGUI format

FreakyFreddie opened the pull request:
**Note: DefectDojo is now on Python3.5 and Django 2.2.x Please submit your pull requests to the 'dev' branch as the 'legacy-python2.7' branch is only for bug fixes. Any new features submitted to the legacy branch will be ignored and closed.**

When submitting a pull request, please make sure you have completed the following checklist:

- [x] Your code is flake8 compliant 
- [x] Your code is python 3.5 compliant (specific python >=3.6 syntax is currently not accepted)
- [x] If this is a new feature and not a bug fix, you've included the proper documentation in the ReadTheDocs documentation folder. https://github.com/DefectDojo/Documentation/tree/master/docs or provide feature documentation in the PR.

Our team is currently testing DefectDojo to follow up with infrastructure-related vulnerabilities in some of our projects. We use both Nessus and Qualys for scanning, but did not manage to import Qualys results with the standard 'Qualys Scan' type. This appears to be using an XML format from the Qualys v1 API, which is not accessible to everyone. I also noticed other people having the same issue (ref: https://github.com/DefectDojo/django-DefectDojo/issues/358), So I decided to port the default webgui XML export format to DefectDojo.

Documentation:
This feature allows users to import the default XML scan export from the Qualys Web interface in DefectDojo.

- [x] Model changes must include the necessary migrations in the dojo/db_migrations folder.
- [x] Add applicable tests to the unit tests.
- [x] Add the proper label to categorize your PR  --> Scan Import

On behalf of DXC Technology.

dr3dd589 commented:
hello @FreakyFreddie,
Thanks for your contribution. Can you please add unittest for it with some sample file like this #2134.
Thanks

FreakyFreddie commented:
I will take this up in 2 weeks.

FreakyFreddie commented:
I added the necessary unit tests and they are working.

Maffooch commented:
Looks great now! Please submit your sample scans to our [sample-scan repo](https://github.com/DefectDojo/sample-scan-files), update the [documentation](https://github.com/DefectDojo/Documentation/blob/master/docs/integrations.rst) of current scanners, and fix merge conflicts. It will be approved shortly afterwards.

FreakyFreddie commented:
Done. documentation & sample file added and fixed a final flake8 compliance issue.

valentijnscholten commented on dojo/fixtures/test_type.json:
```diff
@@ -585,11 +585,18 @@
 		},
 		"model": "dojo.test_type",
-		"pk": 176
+		"pk": 177
```
can you change this so that the pk of existing scanners doesn't change?

valentijnscholten commented on dojo/tools/qualys_infrascan_webgui/parser.py:
```diff
@@ -0,0 +1,213 @@
+try:
+    from . import utfdictcsv
+except ImportError:
+    print("Missing dict to csv converter custom library. utfdictcsv.py should be in the same path as this file.")
```
please use debug logging or no logging at all and return error to the client.

valentijnscholten commented on dojo/tools/qualys_infrascan_webgui/parser.py:
```diff
@@ -0,0 +1,213 @@
+    from lxml import etree
+except ImportError:
+    print("Missing lxml library. Please install using PIP. https://pypi.python.org/pypi/lxml/3.4.2")
+    exit()
```
I don't think exit() should be used in a webapp?

FreakyFreddie commented:
@valentijnscholten: Can you check if it is okay like this?
I changed the print statement to debug logging and removed the exit(). I had actually copied this from the original Qualys parser.py, so I also resolved the issue there.

test_type pk also updated!

valentijnscholten commented:
There was an error in the integration tests around `Test_Type`: https://travis-ci.org/github/DefectDojo/django-DefectDojo/jobs/684434985 I restarted the build which might fix. Or not.

FreakyFreddie commented:
Should be good now.
